### ISSUE TITLE
Updating to androidx.annotation:annotation:1.5.0 is causing CI failures. 

### ISSUE THREAD

**Issue by reidbaker** [flutter/flutter#122039]
Take a look at https://github.com/flutter/flutter/pull/121847#discussion_r1124797112? I think by bumping a dependency on androidx.annotation:annotation:1.5.0, now some of the test apps in the flutter framework cannot do dependency resolution. 

https://github.com/flutter/packages/pull/3271/files#diff-a7a667ae2f4a6b9d0a2fc91cc1eb18310cf7bc98cecbcd9b27fe444028d87887R37

**Comment by gmackall** [comment]
I am not sure, but it looks like we are failing because we updated to androidx.annotation:annotation:1.5.0, and the particular test "deprecated_gradle_settings.dart" is testing settings [the working directory](https://github.com/flutter/flutter/blob/main/packages/flutter_tools/test/integration.shard/deprecated_gradle_settings_test.dart#L16) to one with [these lockfiles ](https://github.com/flutter/flutter/tree/main/dev/integration_tests/gradle_deprecated_settings/android). The one for the particular plugin specifies [androidx.annotation:annotation:1.5.0](https://github.com/flutter/flutter/blob/main/dev/integration_tests/gradle_deprecated_settings/android/project-flutter_plugin_android_lifecycle.lockfile#L6).

[This comment](https://github.com/flutter/flutter/pull/83067#pullrequestreview-665745101) seems to indicate that we may need to update the the particular lockfile first.

**Comment by reidbaker** [comment]
FWIW I ran that command in the packages/flutter_plugin_android_lifecycle/example/android lib and didnt see any changes to versioned files.

**Comment by gmackall** [comment]
Running the command created untracked lockfiles in the `packages/flutter_plugin_android_lifecycle/example/android` for me, with an updated androidx.annotation:annotation version (1.5.0).

**Comment by reidbaker** [comment]
the only location I see we use this dependency is in `/packages/flutter_plugin_android_lifecycle/android/build.gradle` when I try the task from that location I get a task not defined error. 
When I use your location `packages/flutter_plugin_android_lifecycle/example/android` I see no changes in my tree. Maybe today we can get some time and I can see what you see.

**Comment by reidbaker** [comment]
@gmackall and I paired on this and he will be taking it over.
TLDR the think to know about lockfiles appears to be to cd one level above the build.gradle that needs locks generated then to run echo <FOLDER> | dart run  <FLUTTER_HOME>/dev/tools/bin/generate_gradle_lockfiles.dart

This will generate multiple lockfiles but at least for the correct build.gradle file.
Note that this tool appears to work on gradle 7.3.3 and above.

**Comment by gmackall** [comment]
Updating as this has gone for a while without any more info: this is still on my backlog, I have a draft PR to fix but it is currently failing many tests. I've dropped the priority for the moment to p4 to accurately reflect that I am unable to work on it at the moment.
